src/atoms/IndexAtom.jl

```diff
@@ -117,6 +117,14 @@ function Base.getindex(x::AbstractExpr, I::AbstractVector{Bool})
     return [xi for (xi, ii) in zip(x, I) if ii]
 end
 
+function Base.getindex(x::AbstractExpr, I::BitMatrix)
+    return [xi for (xi, ii) in zip(x, I) if ii]
+end
+
+function Base.getindex(x::AbstractExpr, I::BitVector)
+    return [xi for (xi, ii) in zip(x, I) if ii]
```

Comment:
I think we need some boundschecking on these, since we should get an error if `I` is too large or has the wrong shape, but `zip` won't do that

Reply:
I copied the code from the Boolean matrix / vector case.
Fixing it here won't change there.

So maybe get this in and add an issue for bound checking for the Boolean / Bit cases later?

Reply:
ah ok. Maybe we can just change the signature of those to
```julia
Base.getindex(x::AbstractExpr, I::Union{<:AbstractVector{Bool}, <:BitVector})
```
? Then we would only need to add the checking in one place later.

Reply:
I'm OK with anything :-).

Reply:
@ericphanson , Done.

Reply:
By the way, didn't contribute anything real here. All credit is you and @odow which I basically copied his text.
So the credit should be yours.